### ISSUE TITLE
feature - rxify

### ISSUE THREAD

**Issue by felixSchl** [Reactive-Extensions/RxJS#493]
We have methods already to "rxify" single functions using `fromNodeCallback` and `fromPromise` etc.

However, I have seen [this function](https://github.com/petkaantonov/bluebird/blob/master/API.md#promisepromisifyfunction-nodefunction--dynamic-receiver---function) on the [bluebird](https://github.com/petkaantonov/bluebird) library which would change the behaviour of all conformant methods to use promises. I thought _might_ be interesting for rxjs to have as well.

Is this something worth looking into or would it be more suited to a third party module?

**Comment by trxcllnt** [comment]
I agree, this would be great. Would obviate the need for modules like https://github.com/Reactive-Extensions/rxjs-node.

**Comment by L8D** [comment]
Would this be for transforming all callback interfaces in the same form as `fromNodeCallback`? Observables are not really meant to be used for one-time callback actions unlike promises. If you are using them like so enough to justify an `rxifyAll` function then you're probably using RxJS wrong. However, observables are great at modeling/wrapping event emitters, like node streams.

**Comment by trxcllnt** [comment]
@L8D Observables are a perfectly reasonable abstraction over single-value asynchronous sources, if for no other reason than they are side-effect free by default. Promises multicast and aren't cancelable, which makes them unsuitable for a whole host of operations.

**Comment by felixSchl** [comment]
@L8D I see where you are coming from. However, Rx does feature an [AsyncSubject](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/subjects/asyncsubject.md) which is made just for that purpose. Not to get off the road too far, but the way I look at rx, is that it's an abstraction to deal with async code - and so are promises, node style callbacks and there are other approaches. Rx has the advantage of using lists over single values, as it inherits the properties of it and other commonly defined functions over lists apply, such as map and flatmap etc. While I appreciate that events lend themselves particularly well to forming a stream of values, I do not see myself mixing and matching abstractions if I do not have to.
